### PR TITLE
ARUHA-334: add location headers

### PR DESCRIPTION
**Pull request by v1ctor**

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/controller/SubscriptionController.java [hunk 1]
```diff
@@ -93,7 +98,9 @@ public SubscriptionController(final SubscriptionDbRepository subscriptionReposit
             return createSubscription(subscriptionBase, request, client);
         } catch (final DuplicatedSubscriptionException e) {
             try {
-                return new ResponseEntity<>(getExistingSubscription(subscriptionBase), HttpStatus.OK);
+                Subscription existingSubscription = getExistingSubscription(subscriptionBase);
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/controller/SubscriptionController.java]: final

File: src/main/java/org/zalando/nakadi/controller/SubscriptionController.java [hunk 2]
```diff
@@ -93,7 +98,9 @@ public SubscriptionController(final SubscriptionDbRepository subscriptionReposit
...
+                UriComponents path = SUBSCRIPTION_PATH.buildAndExpand(existingSubscription.getId());
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/controller/SubscriptionController.java]: final

File: src/main/java/org/zalando/nakadi/controller/SubscriptionController.java [hunk 3]
```diff
@@ -176,7 +183,11 @@ public SubscriptionController(final SubscriptionDbRepository subscriptionReposit
 
         // generate subscription id and try to create subscription in DB
         final Subscription subscription = subscriptionRepository.createSubscription(subscriptionBase);
-        return status(HttpStatus.CREATED).body(subscription);
+        UriComponents components = SUBSCRIPTION_PATH.buildAndExpand(subscription.getId());
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/controller/SubscriptionController.java]: final

**Comment by v-stepanov** [comment]
It would be nice to have a controller test for that.

**Comment by v-stepanov** [comment]
:shipit:

**Comment by rcillo** [comment]
We also need a separated PR to change the API definition. There it's stated that URI is absolute.

**Pull request merged**